### PR TITLE
[CodeCamp2023-584]Support DINO self-supervised learning in project

### PR DESCRIPTION
**Pull request by LALBJ**
Support DINO self-supervised learning in project, based on [https://github.com/open-mmlab/OpenMMLabCamp/discussions/584](https://github.com/open-mmlab/OpenMMLabCamp/discussions/584)

## Motivation

To support DINO in MMpretrain, migrate DINO from MMSelSup.

## Modification

Add DINO python files in mmpretrain/project

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+model = dict(
+    type='DINO',
+    data_preprocessor=dict(
+        mean=[123.675, 116.28, 103.53],
+        std=[58.395, 57.12, 57.375],
+        bgr_to_rgb=True),
+    backbone=dict(type='mmcls.VisionTransformer', arch='b', patch_size=16),
```
Comment (fangyixiao18) [inline projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py:7]:
```suggestion
    backbone=dict(type='mmpretrain.VisionTransformer', arch='b', patch_size=16),
```
please use model in mmpretrain

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    dict(type='PackInputs', meta_keys=['img_path'])
+]
+train_dataloader = dict(
+    batch_size=32,
+    num_workers=16,
+    persistent_workers=True,
+    sampler=dict(type='DefaultSampler', shuffle=True),
+    collate_fn=dict(type='default_collate'),
+    dataset=dict(
+        type='mmcls.ImageNet',
```
Comment (fangyixiao18) [inline projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py:36]:
```suggestion
        type='mmpretrain.ImageNet',
```

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        data_root='/home/liushi_22151211/imagenet/classification',
```
Comment (fangyixiao18) [inline projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py:37]:
```suggestion
        data_root='data/imagenet/',
```
after your experiment. please change back to original path

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        student_temp=0.1,
+        center_momentum=0.9))
+train_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(
+        type='DINOMultiCrop',
+        global_crops_scale=(0.4, 1.0),
+        local_crops_scale=(0.05, 0.4),
+        local_crops_number=8),
+    dict(type='PackInputs', meta_keys=['img_path'])
```
Comment (fangyixiao18) [inline projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py:27]:
```suggestion
    dict(type='PackInputs')
```

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py (hunk 1 above)
Comment (LALBJ) [reply projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py]: done

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py (hunk 2 above)
Comment (LALBJ) [reply projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py]: done

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py (hunk 3 above)
Comment (LALBJ) [reply projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py]: done

File: projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py (hunk 4 above)
Comment (LALBJ) [reply projects/dino/config/dino_vit-base-p16_8xb64-amp-coslr-100e_in1k.py]: done

**Review by fangyixiao18 — APPROVED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @LALBJ,

We'd like to express our appreciation for your valuable contributions to the mmpretrain. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

**Pull request merged**